herddb-core/src/main/java/herddb/index/blink/BLinkKeyToPageIndex.java

```diff
@@ -489,43 +494,109 @@ public Bytes getPosiviveInfinityKey() {
                         throw new IOException("Wrong block type " + block);
                     }
 
-                    final boolean leaf = edis.readBoolean();
-
-                    long id = edis.readVLong();
-                    long storeId = edis.readVLong();
+                    BLinkNodeMetadata<Bytes> node;
 
-                    boolean empty = edis.readBoolean();
+                    if (version == VERSION_2) {
+                        node = readV2(edis, recalculateSize);
 
-                    int keys = edis.readVInt();
-                    long bytes = edis.readVLong();
+                    } else if (version == VERSION_1) {
+                        node = readV1(edis, recalculateSize);
 
-                    if (recalculateSize) {
-                        /* Set size to unknown to force node size recalculation */
-                        bytes = BLink.UNKNOWN_SIZE;
-                    }
-
-                    long outlink = edis.readZLong();
-                    long rightlink = edis.readZLong();
-
-                    boolean hasInf = edis.readBoolean();
+                    } else if (version == VERSION_0) {
+                        node = readV0(edis, recalculateSize);
 
-                    Bytes rightsep;
-                    if (hasInf) {
-                        rightsep = Bytes.POSITIVE_INFINITY;
                     } else {
-                        rightsep = edis.readBytesNoCopy();
+                        throw new IllegalArgumentException("Unknown BLink node metadata version " + version);
                     }
 
-                    BLinkNodeMetadata<Bytes> node
-                        = new BLinkNodeMetadata<>(leaf, id, storeId, empty, keys, bytes, outlink, rightlink, rightsep);
-
                     nodes.add(node);
                 }
 
                 return new BLinkMetadata<>(nextID, fast, fastheight, top, topheight, first, values, nodes);
             }
 
         }
+
+
+        /**
+         * Read node metadata V2
+         */
+        public BLinkNodeMetadata<Bytes> readV2(ByteArrayCursor cursor, boolean recalculateSize) throws IOException {
+
+            final boolean leaf = cursor.readBoolean();
+
+            long id = cursor.readVLong();
+            long storeId = cursor.readVLong();
+
+            int keys = cursor.readVInt();
+            long bytes = cursor.readVLong();
+
+            if (recalculateSize) {
+                /* Set size to unknown to force node size recalculation */
+                bytes = BLink.UNKNOWN_SIZE;
+            }
+
+            long outlink = cursor.readZLong();
+            long rightlink = cursor.readZLong();
+
+            boolean hasInf = cursor.readBoolean();
+
+            Bytes rightsep;
+            if (hasInf) {
+                rightsep = Bytes.POSITIVE_INFINITY;
+            } else {
+                rightsep = cursor.readBytesNoCopy();
```

Comment:
No really sure.
In v1 was already with *readBytesNoCopy*
Should we perform a copy? (In v1 as well)

Reply:
Interesting, good catch